### PR TITLE
Update text color contrast

### LINKED ISSUE #148: Visual accessibility issues in the remix.run/docs site
### Reproduction

1. Visit https://remix.run/docs/en/main/start/quickstart
2. Run an accessibility audit, such as with [Lighthouse](https://developer.chrome.com/docs/lighthouse/overview) or [axe Devtools](https://www.deque.com/axe/devtools)

### System Info

```shell
n/a
```

### Used Package Manager

npm

### Expected Behavior

Pages such as https://remix.run/docs/en/main/start/quickstart shouldn't have any [web accessibility](https://web.dev/explore/accessible) issues.

### Actual Behavior

Scanning with [axe DevTools](https://www.deque.com/axe/devtools) finds two visual errors:

* [Insufficient color contrast](https://dequeuniversity.com/rules/axe/4.8/color-contrast?application=AxeChrome): folks with low color perception and/or poor hardware or lighting conditions might not be able to read...
    * ...in dark mode: the `#666666` commented-out text (e.g. `# install runtime dependencies`)
    * ...in light mode: the `#30ba2d` light green commands (e.g. `my-remix-app`)
* [Links indistinguishable without relying on color](https://dequeuniversity.com/rules/axe/4.8/link-in-text-block?application=AxeChrome): the color contrast between link colors and regular body text similarly isn't enough for some folks to be able to see the difference...
    * ...in dark mode: between the `#3992ff` link text compared to `#a4a4a4` body text
    * ...in light mode: between `#1b6ef5` link text compared to `#434343` body text

I haven't dug too deeply. There might be more insufficient color contrasts or other issues.

Some quick suggested remediations could be:
* Lightning the comments in dark mode to be more visually distinct
* Darkening the green text in light mode to be more visually distinct
* Adding an underline to link text, the way GitHub recently did for Markdown by default

Note that I'm not an accessibility professional and might have missed some issues. I'd be happy to send in a PR with these fixes or alternatives, but would recommend having a real professional go through a full site audit.

### PR DESCRIPTION
**Pull request by IHIutch**
Fixes #148

Fixes a few other text color related issues I noticed while QA'ing

### CODE REVIEW

**Comment by kentcdodds** [comment]
This is awesome! I'm sure the team would appreciate some before/after screenshots to make this easier to review and merge.

**Comment by IHIutch** [comment]
Homepage before
Screenshot 2024-06-10 at 11 09 19 AM


Homepage after
Screenshot 2024-06-10 at 11 09 11 AM

Light mode docs before
Screenshot 2024-06-10 at 11 10 32 AM

Light mode docs after
Screenshot 2024-06-10 at 11 10 45 AM
0825047/72400115-497c-47ac-9826-0d18b687b783">

Dark mode docs before
Screenshot 2024-06-10 at 11 10 18 AM

Dark mode docs after
Screenshot 2024-06-10 at 11 09 50 AM

Blog before
Screenshot 2024-06-10 at 11 13 30 AM

Blog after
Screenshot 2024-06-10 at 11 13 22 AM
(the color change here was actually unintentional, but now matches the docs)

**Comment by IHIutch** [comment]
In theory, you could take this a step further and remove `antialiased` from the body to help with perceived contrast https://github.com/Myndex/SAPC-APCA/discussions/28

Although there is some suggestion that it can be used in dark mode: https://usabilitypost.com/2012/11/05/stop-fixing-font-smoothing/#please-stop-fixing-webkit-font-smoothing

**Comment by machour** [comment]
Looks great! 

Are the new underlines under links made on purpose? Not sure I like that particular change

File: app/styles/docs.css [hunk 1]
```diff
@@ -179,7 +179,7 @@
   /*****************************************************************************/
   /* links */
   & a {
-    @apply text-blue-600 decoration-1 underline-offset-2 hover:underline dark:text-blue-500;
+    @apply text-blue-700 underline decoration-1 underline-offset-2 hover:underline dark:text-blue-500;
```
Comment (kiliman) [inline app/styles/docs.css:182]: Looks like you added `underline` to all links instead of just `hover`

**Comment by IHIutch** [comment]
Yes, that is intentional. Even with the adjustments to the link color it does not satisfy the distinction between text and link color enough to pass without an underline. https://dequeuniversity.com/rules/axe/html/4.4/link-in-text-block

**Review by sergiodxa — APPROVED** [review]

**Review by machour — APPROVED** [review]

**Comment by brookslybrand** [comment]
@ryanleichty if you have a moment I'd love your eyes on this

**Comment by brookslybrand** [comment]
Just deployed to [staging](https://remixdotrunstage.fly.dev/) for easy review

File: app/styles/docs.css (hunk 1 above)
Comment (ryanleichty) [reply app/styles/docs.css:182]: I'm not personally a fan of the underlines from an aesthetic standpoint in this context. There's plenty of reputable sites (e.g., Apple, Vercel, Joshua Comeau's blog, etc) that only use color to indicate a link. I'll defer to @brookslybrand here

**Review by ryanleichty — COMMENTED** [review]
Overall looks good! I like the color contrast improvements 🎉

The only thing I'm not a fan of personally is the underline links (left a comment elsewhere).

File: app/routes/_marketing._index.tsx [hunk 2]
```diff
@@ -138,7 +138,7 @@ function Sample({ html, className }: { html: string; className?: string }) {
   return (
     <div
       className={
-        "relative overflow-auto bg-gray-800 p-3 text-sm sm:rounded-lg sm:text-base md:max-w-full lg:max-w-max xl:rounded-xl xl:p-4" +
+        "relative overflow-auto bg-[#252525] p-3 text-sm sm:rounded-lg sm:text-base md:max-w-full lg:max-w-max xl:rounded-xl xl:p-4" +
```
[the remark this answers is not in the document]
Comment (IHIutch) [reply app/routes/_marketing._index.tsx:141]: `#252525` is the "midpoint" between gray.800 (`#383838`) and gray.900 (`#121212`), for future reference

https://colorkit.io/#383838-#121212-3

**Comment by sergiodxa** [comment]
Many mentions the underline on links, I just want to say that I'm 👍 to adding them, I always like when links have a proper underline and it's just more accessible, and a11y should be over aesthetics if they can't work together.

**Comment by brookslybrand** [comment]
Yeah, I don't really like the underlines either, and was going to push back, but since these are docs, and so largely textual content, I think the more clear the better. reactrouter.com is also already using underlines, so it also aligns them a little more.

We're likely going to be doing some redesigns at some point anyway, and this overall moves us in a better direction and closed an issue someone kindly opened up ~6 months ago

Thanks @IHIutch for tackling this one! 🎉

File: app/routes/_marketing._index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (brookslybrand) [reply app/routes/_marketing._index.tsx:141]: Yep, I might come back and make the gray-900 this and the gray-950 the current gray-900. That's pretty low on the list so if one of you would like to do that I would not complain 😅

**Pull request merged**